### PR TITLE
Tsk1308: Fixed ldap group query for access ids

### PR DESCRIPTION
**Pull request by holgerhagen**

### CODE REVIEW

**Comment by holgerhagen** [comment]
https://sonarcloud.io/dashboard?branch=tsk1308&id=holgerhagen_taskana

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/AccessIdController.java [hunk 1]
```diff
@@ -58,16 +66,20 @@ public AccessIdController(LdapClient ldapClient) {
...
       throw new InvalidArgumentException("The accessId is invalid");
     }
-    List<AccessIdRepresentationModel> accessIdUsers;
+    List<AccessIdRepresentationModel> accessIds;
     ResponseEntity<List<AccessIdRepresentationModel>> response;
-    accessIdUsers = ldapClient.searchUsersAndGroups(accessId);
-    accessIdUsers.addAll(ldapClient.searchGroupsofUsersIsMember(accessId));
-    response = ResponseEntity.ok(accessIdUsers);
+    accessIds = ldapClient.searchGroupsAccessIdIsMemberOf(accessId);
+    response = ResponseEntity.ok(accessIds);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/AccessIdController.java:82]: Why declare the type and then assign the value? This can be optimized (you don't have to do this.. I just noticed it)

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java [hunk 2]
```diff
@@ -187,15 +188,15 @@ public AccessIdRepresentationModel searchGroupByDn(final String name) {
     return accessId;
   }
 
-  public List<AccessIdRepresentationModel> searchGroupsofUsersIsMember(final String name)
+  public List<AccessIdRepresentationModel> searchGroupsAccessIdIsMemberOf(final String name)
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java:191]: Does this work recursively?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java [hunk 3]
```diff
@@ -187,15 +188,15 @@ public AccessIdRepresentationModel searchGroupByDn(final String name) {
...
-    LOGGER.debug("entry to searchGroupsofUsersIsMember(name = {}).", name);
+    LOGGER.debug("entry to searchGroupsAccessIdIsMemberOf(name = {}).", name);
     isInitOrFail();
     testMinSearchForLength(name);
 
     final AndFilter andFilter = new AndFilter();
-    andFilter.and(new WhitespaceWildcardsFilter(getGroupNameAttribute(), ""));
-    andFilter.and(new EqualsFilter(getGroupsOfUser(), name));
+    andFilter.and(new EqualsFilter(getGroupSearchFilterName(), getGroupSearchFilterValue()));
+    andFilter.and(new LikeFilter(getGroupsOfUser(), "*" + name + "*"));
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java:199]: What's the reason for the like? Are we not searching strictly according to the accessId `name`?

File: rest/taskana-rest-spring/src/test/resources/application.properties [hunk 4]
```diff
@@ -25,13 +25,13 @@ taskana.ldap.userSearchFilterValue=person
 taskana.ldap.userFirstnameAttribute=givenName
 taskana.ldap.userLastnameAttribute=sn
 taskana.ldap.userIdAttribute=uid
-taskana.ldap.groupSearchBase=cn=groups
+taskana.ldap.groupSearchBase=
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/resources/application.properties:28]: What's with the user search base?
And what about our other X configs:
* rest/taskana-rest-spring-example-boot/src/main/resources/application-postgres.properties
* rest/taskana-rest-spring-example-boot/src/main/resources/application.properties
* rest/taskana-rest-spring-example-common/src/test/resources/application.properties
* rest/taskana-rest-spring-example-wildfly/src/main/resources/application-postgres.properties
* rest/taskana-rest-spring-example-wildfly/src/main/resources/application.properties
* rest/taskana-rest-spring-example-wildfly/src/test/resources/application.properties
* rest/taskana-rest-spring/src/test/resources/application.properties
* ADAPTER: taskana-adapter-camunda-wildfly-example/src/main/resources/application.properties

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/AccessIdControllerIntTest.java [hunk 5]
```diff
@@ -110,6 +111,79 @@ void testBadRequestWhenSearchForIsTooShort() {
...
+    ResponseEntity<List<AccessIdRepresentationModel>> response =
+        template.exchange(
+            restHelper.toUrl(Mapping.URL_ACCESSID_GROUPS) + "?access-id=teamlead-2",
+            HttpMethod.GET,
+            restHelper.defaultRequest(),
+            ParameterizedTypeReference.forType(AccessIdListResource.class));
+
+    List<AccessIdRepresentationModel> body = response.getBody();
+    assertThat(body).isNotNull();
+    assertThat(body)
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/AccessIdControllerIntTest.java:125]:
```suggestion
    assertThat(response.getBody()).isNotNull()
```

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/AccessIdControllerIntTest.java [hunk 6]
```diff
@@ -110,6 +111,79 @@ void testBadRequestWhenSearchForIsTooShort() {
...
+            restHelper.toUrl(Mapping.URL_ACCESSID_GROUPS)
+                + "?access-id=cn=Organisationseinheit KSC 1,"
+                + "cn=Organisationseinheit KSC,cn=organisation,OU=Test,O=TASKANA",
+            HttpMethod.GET,
+            restHelper.defaultRequest(),
+            ParameterizedTypeReference.forType(AccessIdListResource.class));
+
+    List<AccessIdRepresentationModel> body = response.getBody();
+    assertThat(body).isNotNull();
+    assertThat(body)
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/AccessIdControllerIntTest.java:149]:
```suggestion
    assertThat(response.getBody()).isNotNull()
```

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 7]
```diff
@@ -476,8 +476,8 @@ void testGetLastPageSortedByDueWithHiddenTasksRemovedFromResult() {
     assertThat(response.getBody()).isNotNull();
     assertThat((response.getBody()).getContent()).hasSize(5);
     assertThat(response.getBody().getRequiredLink(IanaLinkRelations.LAST).getHref())
-        .contains("page=5");
-    assertThat("TKI:000000000000000000000000000000000023")
+        .contains("page=10");
+    assertThat("TKI:000000000000000000000000000000000005")
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:480]: whats the reason for the changes in this file? I don't see a matching change

File: rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java [hunk 8]
```diff
@@ -82,7 +82,7 @@ void testGetAllWorkbasketsBusinessAdminHasOpenPermission() {
             WORKBASKET_SUMMARY_PAGE_MODEL_TYPE);
     assertThat(response.getBody()).isNotNull();
     assertThat(response.getBody().getRequiredLink(IanaLinkRelations.SELF)).isNotNull();
-    assertThat(response.getBody().getContent()).hasSize(3);
+    assertThat(response.getBody().getContent()).hasSize(6);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java:85]: what's the reason for this change?

**Comment by mustaphazorgati** [comment]
Hey @holgerhagen,
additional to the changes I expected something to be in the release notes. Is there a reason you didn't add anything?
The new behaviour that you have to be admin/business admin to use the endpoint is meantionable

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java (hunk 2 above)
Comment (holgerhagen) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java:191]: No, it doesn't. And based on my understanding, it shouldn't. Do you have another opinion?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java (hunk 3 above)
Comment (holgerhagen) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java:199]: No. name could be a user, which we handle as uid olny. In the uniquemember field we do have the DN for users and groups.

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java (hunk 7 above)
Comment (holgerhagen) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:480]: It's due to the changed groupSearchBase. Now he uses the org groups as well, resulting in more hits.

File: rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java (hunk 8 above)
Comment (holgerhagen) [reply rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java:85]: see above.

File: rest/taskana-rest-spring/src/test/resources/application.properties (hunk 4 above)
Comment (holgerhagen) [reply rest/taskana-rest-spring/src/test/resources/application.properties:28]: We can decide about the other properties. They don't need to be exactly the same. Do you prefer to have the same settingseverywhere?

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java (hunk 2 above)
Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java:191]: Was just wondering. Nothing more :)

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java (hunk 3 above)
Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java:199]: Then I don't like the name of the method. The methodname implies that the parameter is a accessId. Can we talk about this? That's faster than typing.

File: rest/taskana-rest-spring/src/test/resources/application.properties (hunk 4 above)
Comment (mustaphazorgati) [reply rest/taskana-rest-spring/src/test/resources/application.properties:28]: Again, let's talk about this :)

Comment (holgerhagen) [reply rest/taskana-rest-spring/src/test/resources/application.properties:28]: Since the properties are different already, I will not touch this point in this ticket.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java [hunk 9]
```diff
@@ -187,15 +188,15 @@ public AccessIdRepresentationModel searchGroupByDn(final String name) {
     return accessId;
   }
 
-  public List<AccessIdRepresentationModel> searchGroupsofUsersIsMember(final String name)
+  public List<AccessIdRepresentationModel> searchGroupsAccessIdIsMemberOf(final String accessId)
       throws InvalidArgumentException {
-    LOGGER.debug("entry to searchGroupsofUsersIsMember(name = {}).", name);
+    LOGGER.debug("entry to searchGroupsAccessIdIsMemberOf(name = {}).", accessId);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/common/rest/ldap/LdapClient.java:193]: name = accessId? Aber egal jetzt :D
